buffer/src/test/java/io/netty/buffer/SlicedByteBufTest.java

```diff
@@ -224,4 +228,26 @@ public void testWriteIso88591CharSequenceExpand() {
     public void testWriteUtf16CharSequenceExpand() {
         super.testWriteUtf16CharSequenceExpand();
     }
+
+    @Test
+    public void ensureWritableWithEnoughSpaceShouldNotThrow() {
+        ByteBuf buf = buffer(1);
+        int originalWriterIndex = buf.writerIndex();
+        buf.writerIndex(buf.writerIndex() + 1);
+        buf.ensureWritable(1);
+        assertEquals(originalWriterIndex + 1, buf.writerIndex());
+        assertEquals(1, buf.readableBytes());
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void ensureWritableWithNotEnoughSpaceShouldThrow() {
+        ByteBuf buf = buffer(1);
+        try {
+            buf.writerIndex(buf.writerIndex() + 2);
```

Comment:
Just use the following code to not depend on the knowledge that we double the requested length of the buffer:

```java
buf.writerIndex(buf.writerIndex() + buf.writableBytes());
```